src/otx/algorithms/action/adapters/mmaction/task.py

```diff
@@ -423,32 +428,29 @@ def hook(module, inp, outp):
         feature_vectors = []
         saliency_maps = []
 
-        def dump_features_hook():
-            raise NotImplementedError("get_feature_vector function for mmaction is not implemented")
-
-        # pylint: disable=unused-argument
-        def dummy_dump_features_hook(model, inp, out):
-            feature_vectors.append(None)
-
-        def dump_saliency_hook():
-            raise NotImplementedError("get_saliency_map for mmaction is not implemented")
-
-        # pylint: disable=unused-argument
-        def dummy_dump_saliency_hook(model, inp, out):
-            saliency_maps.append(None)
-
-        feature_vector_hook = dump_features_hook if dump_features else dummy_dump_features_hook
-        saliency_map_hook = dump_saliency_hook if dump_saliency_map else dummy_dump_saliency_hook
+        if not dump_features:
+            feature_vector_hook: Union[nullcontext, BaseRecordingForwardHook] = nullcontext()
+        else:
+            feature_vector_hook = FeatureVectorHook(feature_model)
+        saliency_hook = nullcontext()
 
         prog_bar = ProgressBar(len(dataloader))
-        with model.module.backbone.register_forward_hook(feature_vector_hook):
-            with model.module.backbone.register_forward_hook(saliency_map_hook):
+        with feature_vector_hook:
+            with saliency_hook:
                 for data in dataloader:
                     with torch.no_grad():
                         result = model(return_loss=False, **data)
                     eval_predictions.extend(result)
                     for _ in range(videos_per_gpu):
                         prog_bar.update()
+                if isinstance(feature_vector_hook, nullcontext):
+                    feature_vectors = [None] * len(mm_dataset)
+                else:
+                    feature_vectors = feature_vector_hook.records
+                if isinstance(saliency_hook, nullcontext):
+                    saliency_maps = [None] * len(mm_dataset)
+                else:
```

Comment:
`saliency_hook` is always `nullcontext()` here, too. will there be additional implementation?

Reply:
613418e15e70b1fe53ecaa3b0707134d52555c7d will handle this